modules/andes-core/client/src/main/java/org/wso2/andes/client/BasicMessageConsumer.java

```diff
@@ -619,7 +619,9 @@ public void close(boolean sendClose) throws JMSException
                         // no point otherwise as the connection will be gone
                         if (!_session.isClosed() || _session.isClosing())
                         {
-                            _session._dispatcher.rejectPending(this);
+                            if(null != _session._dispatcher) {
```

Comment:
See https://github.com/wso2/andes/blob/master/modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java#L2581

We are on same call path.